src/advanced_inv.cpp

```diff
@@ -631,40 +636,26 @@ void advanced_inventory::init()
         square.init();
     }
 
-    panes[left].area = AIM_ALL;
-    panes[right].area = AIM_INVENTORY;
-
-    panes[left].sortby = ( advanced_inv_sortby ) uistate.adv_inv_leftsort;
-    panes[right].sortby = ( advanced_inv_sortby ) uistate.adv_inv_rightsort;
-    panes[left].area = ( aim_location ) uistate.adv_inv_leftarea;
-    panes[right].area = ( aim_location ) uistate.adv_inv_rightarea;
-
     bool moved = ( uistate.adv_inv_last_coords.x != g->u.posx() || 
                    uistate.adv_inv_last_coords.y != g->u.posy() ||
                    uistate.adv_inv_last_coords.z != g->u.posz());
 
-    // finally! if it calls to load it, load it! (it being prior vehicle data)
-    if(panes[left].area == AIM_VEHICLE || panes[right].area == AIM_VEHICLE) {
-        // have we moved since last usage?
-        if(!moved) {
-            load_veh_data();
-        } else {
-            // reset pane that is AIM_VEHICLE
-            if(panes[left].area == AIM_VEHICLE) {
-                panes[left].area = AIM_ALL;
-            }
-            if(panes[right].area == AIM_VEHICLE) {
-                panes[right].area = AIM_INVENTORY;
-            }
-            // disable loading our data to panes[right]event doing this over again
-            uistate.adv_inv_load_veh = false;
-        }
-    }
+    panes[left].sortby  = (advanced_inv_sortby) static_cast<advanced_inv_sortby>(uistate.adv_inv_leftsort);
+    panes[right].sortby = (advanced_inv_sortby) static_cast<advanced_inv_sortby>(uistate.adv_inv_rightsort);
```

Comment:
I see, you want to be really, really sure it's an `advanced_inv_sortby` value, so you cast it with a `static_cast` and additionally with a C-style cast. One never knowns what those fancy static_casts return, right? They could return anything, better make sure it's the correct type (-: